layers/ip4.go

```diff
@@ -110,9 +112,15 @@ func (ip *IPv4) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeO
 	}
 	bytes[0] = (ip.Version << 4) | ip.IHL
 	bytes[1] = ip.TOS
-	binary.BigEndian.PutUint16(bytes[2:], ip.Length)
+	switch runtime.GOOS {
```

Comment:
I'm quite certain that the byte order of an IP packet does not change depending on the host OS.  I believe this code is in error.  Packet format doesn't change per-OS or per-architecture... if it did, different OS's wouldn't be able to communicate with each other over the wire.

Note that SerializeTo is writing the WIRE format for packets, which for IPv4 is always big-endian.

Can you provide a concrete example where this isn't the case?  A captured packet from one of these OS's that shows the little-endian order?